### PR TITLE
Add support for regclass and current_schemas()

### LINKED ISSUE #3416: Support Postgres OID alias types
See the [Postgres docs](https://www.postgresql.org/docs/current/datatype-oid.html)

e.g.

```sql
SELECT * FROM pg_attribute WHERE attrelid = 'mytable'::regclass;

-- effectively expands to

SELECT * FROM pg_attribute
  WHERE attrelid = (SELECT oid FROM pg_class WHERE relname = 'mytable');
```

Note: 

> While that doesn't look all that bad by itself, it's still oversimplified. A far more complicated sub-select would be needed to select the right OID if there are multiple tables named mytable in different schemas. The regclass input converter handles the table lookup according to the schema path setting, and so it does the “right thing” automatically. Similarly, casting a table's OID to regclass is handy for symbolic display of a numeric OID.

### PR DESCRIPTION
**Pull request by mbutlerw**
resolves #3416

doing https://github.com/xtdb/xtdb/pull/3630 first

### CODE REVIEW

File: core/src/main/kotlin/xtdb/operator/IProjectionSpec.kt [hunk 1]
```diff
@@ -12,5 +13,5 @@ interface IProjectionSpec {
     /**
      * @param params a single-row indirect relation containing the params for this invocation - maybe a view over a bigger param relation.
      */
-    fun project(allocator: BufferAllocator, readRelation: RelationReader, params: RelationReader): IVectorReader
+    fun project(allocator: BufferAllocator, readRelation: RelationReader, schema: IPersistentMap, params: RelationReader): IVectorReader
```
Comment (jarohen) [inline core/src/main/kotlin/xtdb/operator/IProjectionSpec.kt:16]: IPM -> `Map`

File: core/src/main/clojure/xtdb/expression/pg.clj [hunk 2]
```diff
@@ -0,0 +1,52 @@
+(ns xtdb.expression.pg
+  (:require [xtdb.expression :as expr]
+            [clojure.string :as str]
+            [xtdb.error :as err]))
+
+(set! *unchecked-math* :warn-on-boxed)
+
+(def search-path
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/expression/pg.clj]: "xt" too?

File: core/src/main/clojure/xtdb/expression/pg.clj [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    (name table)
+    (str (namespace table) "." (name table))))
+
+(defn find-matching-table-name [schema tn]
+  (some #(when (contains? (:tables schema) %) %) (symbol-table-names tn)))
+
+(defmethod expr/codegen-cast [:utf8 :regclass] [{:keys [target-type]}]
+  {:return-type target-type
+   :->call-code
+   #(do
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/expression/pg.clj:27]:
```clojure
(fn [[rc-code]]
  ...)
```

File: core/src/main/clojure/xtdb/expression/pg.clj [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+           (hash matching-tn#)
+           (throw (err/runtime-err ::unknown-relation
+                                   {::err/message (format "Relation %s does not exist" tn#)})))))})
+
+(defmethod expr/codegen-cast [:regclass :utf8] [{:keys [target-type]}]
+  {:return-type target-type
+   :->call-code
+   #(do
+      `(let [oid# ~@%]
+        (if-let [tn# (get-in ~expr/schema-sym [:oid->table oid#])]
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/expression/pg.clj:39]: indentation

File: core/src/main/clojure/xtdb/expression/pg.clj [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          (expr/resolve-utf8-buf (string-table-name tn#))
+          (expr/resolve-utf8-buf (str oid#)))))})
+
+(defmethod expr/codegen-cast [:regclass :int] [{:keys [target-type]}]
+  {:return-type target-type
+   :->call-code #(do `(do ~@%))})
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/expression/pg.clj:45]: `first`

**Review by jarohen — CHANGES_REQUESTED** [review]
* schema moving to watermark, make it consistent
* empty watermark for tests to remove the nil-handling in real code
* zero the first bit of/abs the oid after hashing it to avoid unsigned/signed issues
* create a `current-schemas` fn outside of the codegen (see expr.temporal)

**Comment by mbutlerw** [comment]
Closing in favour off separate PRs which bring in the above changes in a more incremental way.

See https://github.com/xtdb/xtdb/pull/3646 for regclass.

**Pull request closed without merging**